docs: link Analyze API & Text analysis from match query page

Username46786 opened the pull request:
### What
Add a short "See also" that links from the match query docs to:
- {ref}/indices/analyze.html (Analyze API)
- {ref}/analysis.html (Text analysis concepts)

### Why
The match query notes that input text is analyzed; these links help readers jump
to how analysis works and how to inspect it via the Analyze API.

### How
- Edited docs/reference/query-dsl/match-query.asciidoc

Username46786 committed:
docs: link Analyze API & Text analysis from match query page

Username46786 committed:
Changelog

Username46786 committed:
Merge branch 'main' into docs/add-analyze-link-from-match

Commit:
[CI] Update transport version definitions

leemthompo reviewed (changes requested):
Hi there, thx for fixing the docs: 

1. You don't need a changelog entry for docs changes
1. The docs now use markdown not asciidoc syntax. Refer to https://elastic.github.io/docs-builder/syntax/links/.

leemthompo committed:
Merge branch 'main' into docs/add-analyze-link-from-match

Username46786 committed:
Delete docs/changelog/134802.yaml

Username46786 committed:
Docs: add See also links (Analyze API, Text analysis concepts) to match query page

Username46786 committed:
Docs: add "See also" links (Analyze API, Text analysis concepts) to match query page Optional Extended Description nginx Copy code Placed the "See also" links below the introduction paragraph  so readers can easily find related docs on

Placed the "See also" links below the introduction paragraph 
so readers can easily find related docs on text analysis and the Analyze API.
